indexing-hadoop/src/main/java/io/druid/indexer/InputRowSerde.java

```diff
@@ -49,7 +57,166 @@
+          typeHelper = STRING_HELPER;
+          break;
+        case LONG:
+          typeHelper = LONG_HELPER;
+          break;
+        case FLOAT:
+          typeHelper = FLOAT_HELPER;
+          break;
+        case DOUBLE:
+          typeHelper = DOUBLE_HELPER;
+          break;
+        default:
+          throw new IAE("Invalid type: [%s]", dimensionSchema.getValueType());
+      }
+      typeHelperMap.put(dimensionSchema.getName(), typeHelper);
+    }
+    return typeHelperMap;
+  }
+
+  public static class StringIndexSerdeTypeHelper implements IndexSerdeTypeHelper<List<String>>
+  {
+    @Override
+    public ValueType getType()
+    {
+      return ValueType.STRING;
+    }
+
+    @Override
+    public byte[] serialize(Object value)
+    {
+      List<String> values = Rows.objectToStrings(value);
+      ByteArrayDataOutput out = ByteStreams.newDataOutput();
+      try {
+        writeStringArray(values, out);
+        return out.toByteArray();
+      }
+      catch (IOException ioe) {
+        throw new RuntimeException(ioe);
+      }
+    }
+
+    @Override
+    public List<String> deserialize(byte[] bytes)
+    {
+      ByteArrayDataInput input = ByteStreams.newDataInput(bytes);
+      try {
+        return readStringArray(input);
+      }
+      catch (IOException ioe) {
+        throw new RuntimeException(ioe);
+      }
+    }
+  }
+
+  public static class LongIndexSerdeTypeHelper implements IndexSerdeTypeHelper<Long>
+  {
+    @Override
+    public ValueType getType()
+    {
+      return ValueType.LONG;
+    }
+
+    @Override
+    public byte[] serialize(Object value)
+    {
+      return Longs.toByteArray(
+          DimensionHandlerUtils.convertObjectToLong(value, true)
+      );
+    }
+
+    @Override
+    public Long deserialize(byte[] bytes)
+    {
+      return Longs.fromByteArray(bytes);
+    }
+  }
+
+  public static class FloatIndexSerdeTypeHelper implements IndexSerdeTypeHelper<Float>
+  {
+    @Override
+    public ValueType getType()
+    {
+      return ValueType.FLOAT;
+    }
+
+    @Override
+    public byte[] serialize(Object value)
+    {
+      return Ints.toByteArray(
+          Float.floatToIntBits(DimensionHandlerUtils.convertObjectToFloat(value, true))
+      );
+    }
+
+    @Override
+    public Float deserialize(byte[] bytes)
+    {
+      return Float.intBitsToFloat(Ints.fromByteArray(bytes));
+    }
+  }
+
+  public static class DoubleIndexSerdeTypeHelper implements IndexSerdeTypeHelper<Double>
+  {
+    @Override
+    public ValueType getType()
+    {
+      return ValueType.DOUBLE;
+    }
+
+
+    @Override
+    public byte[] serialize(Object value)
+    {
+      return Longs.toByteArray(
+          Double.doubleToLongBits(DimensionHandlerUtils.convertObjectToDouble(value, true))
+      );
+    }
+
+    @Override
+    public Double deserialize(byte[] bytes)
+    {
+      return Double.longBitsToDouble(Longs.fromByteArray(bytes));
+    }
+  }
+
+  public static final byte[] toBytes(
```

Comment:
Please add `@Nullable` and a javadoc explaining what it means when this returns null (probably it means there was a parse exception and `reportParseExceptions` is false? Is it possible for this to return null if `reportParseExceptions` is true?).

Reply:
I think this might never return null if you take one of the other suggestions (skip column instead of skip whole row on parse error).